.jenkins/Jenkinsfile

```diff
@@ -16,6 +16,8 @@ AGENTS_LABELS = [
     "acc-rhel-8":       env.RHEL_8_CUSTOM_LABEL ?: "ACC-RHEL-8",
     "acc-win2016":      env.WINDOWS_2016_CUSTOM_LABEL ?: "SGXFLC-Windows",
     "acc-win2016-dcap": env.WINDOWS_2016_DCAP_CUSTOM_LABEL ?: "SGXFLC-Windows-DCAP",
```

Comment:
do the strings at the end of the line need to match something else, or can "-2016" be added to the strings at end of 17-18 to distinguish them from 19-20?

Reply:
There are two issues I see here.

```
    "acc-win2016":      env.WINDOWS_2016_CUSTOM_LABEL ?: "SGXFLC-Windows",
    "acc-win2016-dcap": env.WINDOWS_2016_DCAP_CUSTOM_LABEL ?: "SGXFLC-Windows-DCAP",
```

I think should be
```
`    "acc-win2016":      env.WINDOWS_2016_CUSTOM_LABEL ?: "**SGXFLC-2016-Windows**",
    "acc-win2016-dcap": env.WINDOWS_2016_DCAP_CUSTOM_LABEL ?: "**SGXFLC-2016-Windows-DCAP**",
```

And secondly lines 19-20 

   ```
 "acc-win2019":      env.WINDOWS_2016_CUSTOM_LABEL ?: "SGX-Windows-2019",
    "acc-win2019-dcap": env.WINDOWS_2016_DCAP_CUSTOM_LABEL ?: "SGXFLC-Windows-2019-DCAP",
```

Should be

   ```
 "acc-win2019":      env.**WINDOWS_2019_CUSTOM_LABEL** ?: "SGX-Windows-2019",
    "acc-win2019-dcap": env.**WINDOWS_2019_DCAP_CUSTOM_LABEL** ?: "SGXFLC-Windows-2019-DCAP",
```

Reply:
Yes thank you for this observation , I fixed the variables names.

However , now that we changed the labels for the Windows 2016 nodes ( Adding 2016 to the label name ) we need to update all the Jenkins VM plugins configs to use the new label.
Those changes need to happen before we merge this PR , otherwise the build will never succeed since there will be no nodes matching   "SGXFLC-2016-Windows" and "SGXFLC-2016-Windows-DCAP"

Reply:
So, the whole dict `AGENTS_LABELS` is used for the Jenkins agents labels. Based on these labels, the Jenkins master spawns Azure machines as configured in the global config.

The instruction (encountered for each label)
```
"acc-ubuntu-16.04": env.UBUNTU_1604_CUSTOM_LABEL ?: "ACC-1604
```
is basically a short-if in Groovy scripting (called [Elvis operator](https://groovy-lang.org/operators.html#_elvis_operator)).

If `env.UBUNTU_1604_CUSTOM_LABEL` is set, then use that, otherwise use the default label `ACC-1604`.

And everywhere in the Jenkinsfile when a `node` primitive is used, you would see the dictionary `AGENTS_LABELS` to find the proper label.

We wanted to make sure we can overwrite the default agents labels via environment variables to re-use the same Jenkinsfile in our end-to-end testing job.

@BRMcLaren I would suggest that we don't use any wildcard expression since we want to make sure that Jenkins master uses the proper Azure VM template for a label, and not accidentally match another label.

@dthaler Regarding your observation about the default labels naming:

Windows Server 2016 with: `SGXFLC-Windows` / `SGXFLC-Windows-DCAP`
Windows Server 2019 with: `SGX-Windows-2019` / `SGX-Windows-2019-DCAP`

The label string is just a Jenkins config that can be changed for the Jenkins Azure plugin, and make it consistent. We can do that. We previously didn't append any version to the label because we had only a Windows Server 2016 supported.

Reply:
Tsk, those asterisks were actually me bolding an item to add emphasis.

Which seemed to have not transferred well. Yes, I agree, we shouldn't use wild cards operators in labels.